### PR TITLE
Remove remaining deprecated props

### LINKED ISSUE #2019: Remove all deprecated props
I'm still seeing things like this in the 2.0.0-beta.1 docs:

image

### PR DESCRIPTION
**Pull request by adidahiya**
#### Fixes #2019

- see commits
- needs a regression pass

### CODE REVIEW

File: packages/docs-app/src/components/blueprintDocs.tsx [hunk 1]
```diff
@@ -78,7 +78,7 @@ export class BlueprintDocs extends React.Component<IBlueprintDocsProps, { themeN
 
         const match = /docs\/v([0-9]+)/.exec(location.href);
         // default to latest release if we can't find a major version in the URL
-        const currentRelease = match == null ? versions[0].version : match[1];
+        const currentRelease = match == null ? versions[versions.length - 1].version : match[1];
```
Comment (giladgray) [inline packages/docs-app/src/components/blueprintDocs.tsx]: 👌

File: packages/docs-theme/src/components/documentation.tsx [hunk 2]
```diff
@@ -121,11 +121,11 @@ export class Documentation extends React.PureComponent<IDocumentationProps, IDoc
             <div className={classNames("docs-root", { "docs-examples-only": examplesOnly }, this.props.className)}>
                 <div className="docs-app">
                     <div className="pt-navbar docs-navbar docs-flex-row">
-                        <div className="pt-navbar-group">{this.props.navbarLeft}</div>
+                        <div className="pt-navbar-group pt-navbar-group-left">{this.props.navbarLeft}</div>
                         <div className="pt-navbar-group">
                             <Navigator items={nav} onNavigate={this.handleNavigation} />
                         </div>
-                        <div className="pt-navbar-group">{this.props.navbarRight}</div>
+                        <div className="pt-navbar-group pt-navbar-group-right">{this.props.navbarRight}</div>
```
Comment (giladgray) [inline packages/docs-theme/src/components/documentation.tsx]: we have components for `Navbar` etc, but whatever

File: packages/table/src/common/errors.ts [hunk 3]
```diff
@@ -5,16 +5,7 @@
...
-
-export const COLUMN_HEADER_CELL_MENU_DEPRECATED =
-    deprec + ` <ColumnHeaderCell> menu is deprecated. Use menuRenderer instead.`;
-export const COLUMN_HEADER_CELL_USE_INTERACTION_BAR_DEPRECATED =
-    deprec +
-    ` <ColumnHeaderCell> enableColumnInteractionBar is deprecated. Pass the prop to the top-level <Table> instead. When you pass the prop to <Table>, that value will override the one provided directly to <ColumnHeaderCell>.`;
-
-export const ROW_HEADER_CELL_MENU_DEPRECATED =
-    deprec + ` <RowHeaderCell> menu is deprecated. Use menuRenderer instead.`;
+// const deprec = `${ns} DEPRECATION:`;
```
Comment (giladgray) [inline packages/table/src/common/errors.ts]: commented code

File: packages/table/src/headers/columnHeaderCell.tsx [hunk 4]
```diff
@@ -85,11 +72,14 @@ export function HorizontalCellDivider(): JSX.Element {
...
 export class ColumnHeaderCell extends AbstractPureComponent<IColumnHeaderCellProps, IColumnHeaderCellState> {
     public static defaultProps: IColumnHeaderCellProps = {
-        enableColumnInteractionBar: false,
         isActive: false,
         menuIconName: "chevron-down",
     };
 
+    public static contextTypes: React.ValidationMap<
+        IColumnInteractionBarContextTypes
+    > = columnInteractionBarContextTypes;
```
Comment (giladgray) [inline packages/table/src/headers/columnHeaderCell.tsx]: shouldn't need `React.ValidationMap...` type info. let it come from `columnInteractionBarContextTypes`

File: packages/table/src/table.tsx [hunk 5]
```diff
@@ -425,6 +422,10 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         selectionModes: SelectionModes.ALL,
     };
 
+    public static childContextTypes: React.ValidationMap<
+        IColumnInteractionBarContextTypes
+    > = columnInteractionBarContextTypes;
```
Comment (giladgray) [inline packages/table/src/table.tsx]: `public static childContextTypes = columnInteractionBarContextTypes` should do it.

File: packages/table/src/common/errors.ts (hunk 3 above)
Comment (adidahiya) [reply packages/table/src/common/errors.ts]: eh, sort of on purpose. will be useful as a reminder for future deprecation messages 🤷‍♂️

File: packages/table/src/headers/columnHeaderCell.tsx (hunk 4 above)
Comment (adidahiya) [reply packages/table/src/headers/columnHeaderCell.tsx]: you do need it. there's a whole "typescript cannot name this type even though it knows where it comes from" rabbit hole which I won't bother explaining here

File: packages/table/src/table.tsx (hunk 5 above)
Comment (adidahiya) [reply packages/table/src/table.tsx]: nope, see above

File: packages/table/src/headers/columnHeaderCell.tsx (hunk 4 above)
Comment (giladgray) [reply packages/table/src/headers/columnHeaderCell.tsx]: dang i gotcha

**Pull request merged**